client/graphite/client.go

```diff
@@ -111,3 +114,25 @@ func (c *Client) String() string {
 	// TODO: add more stuff here.
 	return c.cfg.String()
 }
+
+// Fetch graphite prefix
+func (c *Client) fetchGraphitePrefix(logger log.Logger, r *http.Request) (string, error) {
+	urlValues, err := url.ParseQuery(r.URL.RawQuery)
+	if err != nil {
+		level.Warn(logger).Log("err", err, "msg", "Error parsing raw URL query")
+		return "", err
+	}
+
+	if graphitePrefix, ok := urlValues["graphite.default-prefix"]; ok {
+		graphitePrefix = urlValues["graphite.default-prefix"]
```

Comment:
why do you assign it to graphitePrefix ?

Reply:
I test if "graphite.default-prefix" exists and then assign it to graphitePrefix, If it doesn't it will use the default prefix set in the config.

Reply:
you don't need to set graphitePrefix in the first branch, and in the second branch you can just return c.cfg.DefaultPrefix